src/timepicker/timepicker.component.ts

```diff
@@ -175,33 +189,90 @@ export class TimepickerComponent
   }
 
   changeHours(step: number, source: TimeChangeSource = ''): void {
+    this.resetValidation();
     this._store.dispatch(this._timepickerActions.changeHours({ step, source }));
   }
 
   changeMinutes(step: number, source: TimeChangeSource = ''): void {
+    this.resetValidation();
     this._store.dispatch(
       this._timepickerActions.changeMinutes({ step, source })
     );
   }
 
   changeSeconds(step: number, source: TimeChangeSource = ''): void {
+    this.resetValidation();
     this._store.dispatch(
       this._timepickerActions.changeSeconds({ step, source })
     );
   }
 
   updateHours(hours: string): void {
+    this.resetValidation();
     this.hours = hours;
+
+    const isValid = isHourInputValid(this.hours, this.isPM())
```

Comment:
isInputLimitValid is called 3 times with the same parameters in this component, maybe it's better to make a function from it.